janitor/timeseries.py

```diff
@@ -176,3 +180,273 @@ def sort_timestamps_monotonically(
+            + f"Must be one of: {direction_types}."
+        )
+
+    if threshold < 0:
+        raise JanitorError(
+            f"Unrecognized threshold: {threshold}. "
+            + "This value must be >= 0.0. "
+            + "Use 'direction' to specify positive or negative intent."
+        )
+
+    single_col = df[col]
+    single_col_diffs = single_col.diff()
+
+    if scale == "percentage":
+        single_col_pcts = single_col.pct_change()
+
+        if direction == "increasing":
+            # Using diffs ensures correct sign is used for incr/decr
+            # (see issue #711)
+            out = (single_col_diffs > 0) & (single_col_pcts.abs() > threshold)
+
+        elif direction == "decreasing":
+            # Using diffs ensures correct sign is used for incr/decr
+            # (see issue #711)
+            out = (single_col_diffs < 0) & (single_col_pcts.abs() > threshold)
+
+        else:
+            out = single_col_pcts.abs() > threshold
+
+    else:
+        if direction == "increasing":
+            out = single_col_diffs > threshold
+
+        elif direction == "decreasing":
+            out = (single_col_diffs < 0) & (single_col_diffs.abs() > threshold)
+
+        else:
+            out = single_col_diffs.abs() > threshold
+
+    out = out.astype(int)
+
+    return out
+
+
+@pf.register_dataframe_method
+def flag_jumps(
+    df: pd.DataFrame,
+    scale: Union[str, Dict[str, str]] = "percentage",
+    direction: Union[str, Dict[str, str]] = "any",
+    threshold: Union[int, float, Dict[str, Union[int, float]]] = 0.0,
+    strict: bool = False,
+) -> pd.DataFrame:
+    """
+    Create boolean column(s) that flag whether or not the change
+    between consecutive rows exceeds a provided threshold.
+
+    Functional usage example:
+
+    .. code-block:: python
+
+        import pandas as pd
+        import janitor.timeseries
+
+        df = pd.DataFrame(...)
+
+        df = flag_jumps(
+            df=df,
+            scale="absolute",
+            direction="any",
+            threshold=2,
+        )
+
+    Method chaining example:
+
+    .. code-block:: python
+
+        import pandas as pd
+        import janitor.timeseries
+
+        df = (
+            pd.DatFrame(...)
+            .flag_jumps(
+                scale="absolute",
+                direction="any",
+                threshold=2,
+            )
+        )
+
+    Detailed chaining examples:
+
+    .. code-block:: python
+
+        # Applies specified criteria across all columns of the dataframe
+        # Appends a flag column for each column in the dataframe
+        df = (
+            pd.DataFrame(...)
+            .flag_jumps(
+                scale="absolute",
+                direction="any",
+                threshold=2
+            )
+        )
+
+        # Applies specific criteria to certain dataframe columns
+        # Applies default criteria to columns not specifically listed
+        # Appends a flag column for each column in the dataframe
```

Comment:
@ericmjl 
I know these comments are a bit cumbersome but I wanted to explicitly tell users the expected behavior. Any recommendations on this? Same applies to the below blocks of comments...

Reply:
No problem here, more comments are better than fewer.